plugins/module_utils/foreman_helper.py

```diff
@@ -357,20 +358,31 @@ def _delete_entity(self, resource, current_entity, params):
         payload = {'id': current_entity['id']}
         if params:
             payload.update(params)
-        self._resource_action(resource, 'destroy', payload)
+        self.resource_action(resource, 'destroy', payload)
         return True, None
 
-    def _resource_action(self, resource, action, params):
+    def resource_action(self, resource, action, params, **kwargs):
         resource_payload = self.foremanapi.resource(resource).action(action).prepare_params(params)
         try:
             result = None
             if not self.check_mode:
-                result = self.foremanapi.resource(resource).call(action, resource_payload)
+                result = self.foremanapi.resource(resource).call(action, resource_payload, **kwargs)
         except Exception as e:
             self.fail_json(msg='Error while performing {} on {}: {}'.format(
                 action, resource, str(e)))
         return True, result
 
+    def wait_for_task(self, task, duration=60, poll=4):
```

Comment:
would this be more readable?

```python
    def wait_for_task(self, task, duration=60, poll=4):
        while duration > 0:
            if task['state'] not in ['paused', 'stopped']:
                time.sleep(poll)
                duration -= poll
                _, task = self.resource_action('foreman_tasks', 'show', {'id': task['id']})
            else:
                break
        else:
            self.fail_json(msg="Timout waiting for Task {}".format(task['id']))
        return task
```

or maybe even

```python
    def wait_for_task(self, task, duration=60, poll=4):
        while duration > 0:
            if task['state'] not in ['paused', 'stopped']:
                time.sleep(poll)
                duration -= poll
                _, task = self.resource_action('foreman_tasks', 'show', {'id': task['id']})
            else:
                return task
        self.fail_json(msg="Timout waiting for Task {}".format(task['id']))
```

Reply:
The tail is not very nice. You do a show but you never check the result which is inefficient. So how about:

```python
def wait_for_task(self, task, duration=60, poll=4):
    while task['state'] not in ['paused', 'stopped']:
        duration -= poll
        if duration <= 0:
            self.fail_json(msg="Timout waiting for Task {}".format(task['id']))
        time.sleep(poll)

        _, task = self.resource_action('foreman_tasks', 'show', {'id': task['id']})

    return task
```

Reply:
I like yours, applied